File: Searching_Algorithms/Python/binary_search.py
def binary_search(arr, target):
    """
    Perform binary search on the given sorted list.

    Parameters:
    - arr: The sorted list to search.
    - target: The element to search for.

    Returns:
    - index: The index of the target if found, otherwise -1.
    """
    low, high = 0, len(arr) - 1

    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1

    return -1

# Example usage:
my_sorted_list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
target_value = 7
result = binary_search(my_sorted_list, target_value)
print(f"Binary Search: Target {target_value} {'found' if result != -1 else 'not found'} at index {result}")

File: Searching_Algorithms/C/binarySearch.c
#include <stdio.h>

int main()
{
    int  i, low, high, mid, n, found = 0, size;
    printf("Enter the size of the array:");
    scanf("%d", &size);
    int a[size];
    printf("Binary Search:\n");
    printf("\nEnter the elements of the array: ");
    for (i = 0; i < size; i++)
    {
        scanf("%d", &a[i]);
    }

    printf("\nEnter the element to be searched: ");
    scanf("%d", &n);

    low = 0;
    high = size - 1;
    while (high >= low)
    {
        mid = (low + high) / 2;
        if (n == a[mid])
        {
            found = 1;
            break;
        }
        else if (n < a[mid])
        {
            high = mid - 1;
        }
        else
        {
            low = mid + 1;
        }
    }
    if (found == 1)
    {
        printf("%d is present in index %d.\n", n, mid);
    }
    else
    {
        printf("%d is not present in the array.\n", n);
    }
    return 0;
}
